src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java

```diff
@@ -386,6 +397,22 @@ public void validate(FailureCollector collector) {
       }
     }
 
+    //This method validated the pattern of CMEK Key resource ID.
+    private void validateCmekKey(FailureCollector failureCollector) {
```

Comment:
this method only validates the CMEK key from the config and not the key from the arguments. It should be validating whatever key will be used (should pass in the key)

Reply:
validateCmekKey() method is called when we click on the validate button in the plugin so it will be concerned about the cmek Key that is passed as config and not the arguments one.